src/lib/services/feature-toggle-service.ts

```diff
@@ -849,8 +850,10 @@ class FeatureToggleService {
                     environment,
                 );
                 if (strategies.length === 0) {
-                    throw new InvalidOperationError(
-                        'You can not enable the environment before it has strategies',
+                    await this.createStrategy(
```

Comment:
this actually bypasses the built-in permission requirement for being allowed to add a strategy in this environment.
I will add a line validating the correct permission.